src/libertem/io/dataset/k2is.py

```diff
@@ -635,7 +655,7 @@ def get_cache_key(self):
     def get_diagnostics(self):
         with self._get_syncer().sectors[0] as sector:
             est_num_frames = sector.filesize // BLOCK_SIZE // BLOCKS_PER_SECTOR_PER_FRAME
```

Comment:
This could now use the new `_get_frame_count` method, right? Or is `est_num_frames` significantly different from `_get_frame_count`?

Reply:
`est_num_frames` is the total number of frames before synchronization and `_get_frame_count` is the number of useful frames after sync, if I understand correctly.

Reply:
Ah right :+1:

Reply:
For the time series data I have, `est_num_frames` and `_get_frame_count` differ by 1 frame. But in regular K2 IS data, there's a large difference because of many blocks without the shutter active flag set, if I understand correctly. I haven't fully understood why there's a 1 frame difference but I'm looking into `sync_sectors`'s 'have_overlap, finding next frame' part.